extensions/amp-a4a/0.1/signature-verifier.js

```diff
@@ -0,0 +1,357 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {signingServerURLs} from '../../../ads/_a4a-config';
+import {cryptoFor} from '../../../src/crypto';
+import {dev} from '../../../src/log';
+import {xhrFor} from '../../../src/services';
+
+/**
+ * Potential reasons why an attempt to verify a Fast Fetch signature might not
+ * succeed. Used for reporting errors to the ad network.
+ *
+ * @enum {number}
+ */
+export const VerificationFailure = {
+
+  /**
+   * A network connectivity failure, misbehaving signing service, or other
+   * factor beyond the ad network's control caused verification to fail.
+   */
+  NO_FAULT: 1,
```

Comment:
This isn't necessarily the signing service's fault; it could also be caused by the user losing network connectivity.